src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -202,12 +202,13 @@ func (c *controller) WaitAndUpdateNodesStatus(ctx context.Context, wg *sync.Wait
 	go c.ApproveCsrs(approveCtx)
 
 	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
-	_ = utils.WaitForPredicateWithContext(ctx, LongWaitTimeout, GeneralWaitInterval, c.waitAndUpdateNodesStatus)
+	_ = utils.WaitForPredicateParamsWithContext(ctx, LongWaitTimeout, GeneralWaitInterval, c.waitAndUpdateNodesStatus, removeUninitializedTaint)
 }
 
-func (c *controller) waitAndUpdateNodesStatus() bool {
+func (c *controller) waitAndUpdateNodesStatus(arg interface{}) bool {
```

Comment:
```suggestion
func (c *controller) waitAndUpdateNodesStatus(removeUninitializedTaint bool)) bool {
```

Reply:
it has to be `arg interface{}` since we're calling it via `utils.WaitForPredicateParamsWithContext` (see similar `waitForCSVBeCreated`)

Reply:
Not better to have as different go routine? or we want to block node from being Done in assisted till we don't remove this taint?

Reply:
That'd increase a number of Nodes.Get() calls, is it really necessary?

The node won't reach Ready status until the taint is not removed (usually cloud-controller removes it but we have fake creds and have to do it ourselves)